jormungandr/src/network/client/mod.rs

```diff
@@ -177,13 +177,14 @@ impl Client {
             // Ignoring possible Pending return here: due to the following
             // ready!() invocations, this function cannot return Continue
             // while no progress has been made.
-            block_sink.as_mut().poll_flush(cx).map_err(|e| {
-                error!(
+            match block_sink.as_mut().poll_flush(cx) {
+                Poll::Ready(Err(e)) => error!(
                     logger,
                     "processing of incoming block messages failed";
                     "reason" => %e,
-                );
-            })?;
```

Comment:
The error is bubbled up with `?` to terminate the client connection. In the proposed change it would be ignored after logging.

Reply:
You can return `Err` at the end of the `Ready(Err(_))` arm, and add a match arm for `Poll::Pending` with a do-nothing body, to where you can move the comment above.

Reply:
Thanks, I misread the code intentions there.